tools/hyperref-check.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+# Extracting all broken hyperref detected by PDFTex
+for file in "./main/acle.md" "./morello/morello.md" "./mve_intrinsics/mve.md" "./neon_intrinsics/advsimd.md"; do
+  echo "Checking $file..."
+
+  # The following string of pipes extracts the unresolved internal links by:
+  # - Using the error and warning messages generated by pandoc pdf conversion as the erroutput variable value
+  #   Note: A Variable is used instead of a pipe in order to avoid EOF errors occuring due to the -zoP flag.
+  # - Using grep to extract all unresolved links, eg "pdfTeX warning (dest): name{ssec-bf16-scalar}"
+  #   Note: Some broken links may be on mutiple lines. The -zoP flag is used to accomodate this. More info here:
+  #   https://unix.stackexchange.com/a/361707
+  # - Using sed to output all the warnings on separate lines to allow them to be counted
+  erroutput=`pandoc $file --verbose -o pdfs/tmp.pdf 2>&1`
+  broken_refs=`grep -zoP 'pdfTeX warning \(dest\): name' <<< "$erroutput" | sed -E  's/name/\n/g'`
+
+  # The -n option was added to prevent the adding of a newline character at the end of
+  # the echo output, which wc counts as a line and thus the script throws an error
+  # when there are no unresolved links.
+  number_of_broken_refs=`echo -n "$broken_refs" | wc -l`
```

Comment:
Now that we try to keep is simple, I suspect we can just do the following:

```suggestion
  number_of_broken_refs=`pandoc $file --verbose -o pdfs/tmp.pdf 2>&1| grep 'pdfTeX warning \(dest\): name' | wc -l`
```

Reply:
```
# Note: A Variable is used instead of a pipe in order to avoid EOF errors occuring due to the -zoP flag.
```
I believe this is here for a reason

Reply:
Actually, I don't think we need that flag anymore

Reply:
Correction: We do.

Reply:
Even with just this invocation? 
```
number_of_broken_refs=`pandoc $file --verbose -o pdfs/tmp.pdf 2>&1| grep 'pdfTeX warning \(dest\): name' | wc -l`
```

Reply:
Oh ok - so we still definitely need the flag but we don't need the extra variable. Also we still need sed in order to get all the warnings on separate lines.